### ISSUE TITLE
Layout/MultilineMethodCallIndentation: one more glitch with long method chains

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#15122]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/MultilineMethodCallIndentation

### Steps to reproduce the problem

```ruby
# frozen_string_literal: true

users
  .dup.sort_by { _1.name.lower } # existance of block here is important to reproduce
  .page(params[:page])
  .per(PER_PAGE)
```

### Expected behavior

The code should be considered correct.

### Actual behavior

```
try_rubocop.rb:5:3: C: [Correctable] Layout/MultilineMethodCallIndentation: Align .page with .sort_by on line 4.
  .page(params[:page])
  ^^^^^
```
Suggested autocorrection:

```ruby
# frozen_string_literal: true

users
  .dup.sort_by { _1.name.lower }
      .page(params[:page])
  .per(PER_PAGE)
```
Again, like in #15120, not only it autocorrects the line weirdly compared to the line above (at least it has _some_ logic), it also leaves the line below completely inconsistent.

This one seems to not related to "autocorrect in hashes" (which #15120/#15054 seem to be caused by)

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ rubocop -V
1.84.2 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 3.1, running on ruby 3.3.7) [x86_64-linux]

(and any version since that, including the current master)
```

**Comment by hammadxcm** [comment]
I've opened a fix for this in #15123.

The root cause was in `handle_descendant_block` and `find_continuation_node` — when the receiver is a single-line block (e.g., `.sort_by { _1.name.lower }`), the cop returned the block's method (`.sort_by`) as the alignment base without considering chained calls before it on the same line (`.dup`). The fix adds a `leftmost_call_on_same_line` helper that walks up through chained receivers to find the correct alignment reference.
